util/metrics/src/lib.rs

```diff
@@ -0,0 +1,22 @@
+use std::time::{Duration, Instant};
+
+pub use metrics::{self as internal, SetRecorderError};
+
+pub struct Timer(Instant);
+
+impl Timer {
+    pub fn start() -> Self {
+        Self(Instant::now())
+    }
+
+    pub fn stop(self) -> Duration {
+        Instant::now() - self.0
+    }
+}
+
+#[macro_export(local_inner_macros)]
+macro_rules! metrics {
```

Comment:
Please add some documents for the metric, at least list the available metrics types.

Reply:
DONE!